src/rebar3_hex_publish.erl

```diff
@@ -114,6 +114,8 @@ format_error(no_write_key) ->
     "No write key found for user. Be sure to authenticate first with:"
     ++ " rebar3 hex user auth";
 
+format_error({publish, {error, {tarball, too_big}}}) -> 
+    "Overall size of package exceeds max allowed size (10MB)"; 
```

Comment:
Message could probably be better.

Reply:
Side note: The max size in this message is not correct. It's actually more like 8.3 megabytes for the tarball and 67 megabytes for the max uncompressed size.

Reply:
might put it as `"Overall size of package exceeds max allowed size (8MB compressed, 67MB total)";` until we can get more accurate values. It won't error out if we're right above the 8mb size either but will clearly explain things if you go above. Whereas here you could get a 9mb package and a confusing error message.

Reply:
Yea, that'd be better.

Reply:
So think I'm about to do a PR up to hex_core to return the size and add a format_error clause on the hex_core end, this way we don't have to worry about the values changing. Thus, I think that should go in, get hex_core to do a release, update this PR etc. Thoughts?

Reply:
Yea, that'd be better.

Reply:
https://github.com/hexpm/hex_core/issues/100

Reply:
Second thought, let's just go with this for now. The PR to hex_core is in, but don't want to rush hex_core on a new release nor have to do a lot of QA per the bigger changes that just went into hex_core.

Reply:
Strike the above, they did a release.

Reply:
PR updated.